exchanges/deribit/deribit.go

```diff
@@ -0,0 +1,2797 @@
+	case kline.ThreeHour:
+		return "180", nil
+	case kline.SixHour:
+		return "360", nil
+	case kline.TwelveHour:
+		return "720", nil
+	case kline.OneDay:
+		return "1D", nil
+	case kline.Raw:
+		return interval.String(), nil
+	default:
+		return "", kline.ErrUnsupportedInterval
+	}
+}
+
+// GetVolatilityIndex gets volatility index for the requested currency
+func (d *Deribit) GetVolatilityIndex(ctx context.Context, ccy, resolution string, startTime, endTime time.Time) ([]VolatilityIndexData, error) {
+	if ccy == "" {
+		return nil, fmt.Errorf("%w '%s'", errInvalidCurrency, ccy)
+	}
+	err := common.StartEndTimeCheck(startTime, endTime)
+	if err != nil {
+		return nil, err
+	}
+	if resolution == "" {
+		return nil, errResolutionNotSet
+	}
+	params := url.Values{}
+	params.Set("currency", ccy)
+	params.Set("start_timestamp", strconv.FormatInt(startTime.UnixMilli(), 10))
+	params.Set("end_timestamp", strconv.FormatInt(endTime.UnixMilli(), 10))
+	params.Set("resolution", resolution)
+	var resp VolatilityIndexRawData
+	err = d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL,
+		common.EncodeURLValues(getVolatilityIndex, params), &resp)
+	if err != nil {
+		return nil, err
+	}
+	response := make([]VolatilityIndexData, len(resp.Data))
+	for x := range resp.Data {
+		response[x] = VolatilityIndexData{
+			TimestampMS: time.UnixMilli(int64(resp.Data[x][0])),
+			Open:        resp.Data[x][1],
+			High:        resp.Data[x][2],
+			Low:         resp.Data[x][3],
+			Close:       resp.Data[x][4],
+		}
+	}
+	return response, nil
+}
+
+// GetPublicTicker gets public ticker data of the instrument requested
+func (d *Deribit) GetPublicTicker(ctx context.Context, instrument string) (*TickerData, error) {
+	params, err := checkInstrument(instrument)
+	if err != nil {
+		return nil, err
+	}
+	var resp *TickerData
+	return resp, d.SendHTTPRequest(ctx, exchange.RestFutures, nonMatchingEPL, common.EncodeURLValues(getTicker, params), &resp)
+}
+
+// SendHTTPRequest sends an unauthenticated HTTP request
+func (d *Deribit) SendHTTPRequest(ctx context.Context, ep exchange.URL, epl request.EndpointLimit, path string, result interface{}) error {
+	endpoint, err := d.API.Endpoints.GetURL(ep)
+	if err != nil {
+		return err
+	}
+	data := &struct {
+		JSONRPC string      `json:"jsonrpc"`
+		ID      int64       `json:"id"`
+		Data    interface{} `json:"result"`
+	}{
+		Data: result,
+	}
+	return d.SendPayload(ctx, epl, func() (*request.Item, error) {
+		return &request.Item{
+			Method:        http.MethodGet,
+			Path:          endpoint + deribitAPIVersion + "/" + path,
+			Result:        data,
+			Verbose:       d.Verbose,
+			HTTPDebugging: d.HTTPDebugging,
+			HTTPRecording: d.HTTPRecording,
+		}, nil
+	}, request.UnauthenticatedRequest)
+}
+
+// GetAccountSummary gets account summary data for the requested instrument
+func (d *Deribit) GetAccountSummary(ctx context.Context, ccy string, extended bool) (*AccountSummaryData, error) {
+	if ccy == "" {
```

Comment:
We can also test for this offline despite not having test credentials set, then once the offline tests have been done we can use the usual `sharedtestvalues.SkipTestIfCredentialsUnset(t, d)`. Noticed a lot of these, so will stop reporting on them